### DISCUSSION TITLE
Converting tif image with clipping path to png

### DISCUSSION THREAD

**Discussion by stigsnor** [ImageMagick/ImageMagick#2061] [Help] [answered]
I'm struggling with converting a tif image with a clipping path to a png and making the area outside of clipping path transparent.

The original tif:
https://drive.google.com/file/d/1ZyK4emowxEooqcn_lnVK_TQCXafSczvc/view?usp=sharing

After googling around and checking forums I'm trying to achieve this using the following command on ImageMagick 7.0.10-13 Q16 x64 2020-05-17 on a Windows 10 machine.

**magick original.tif -alpha transparent -clip -alpha opaque output.png**

Viewing the image after the operation I can seen that an alpa channel has been added, but the area outside of clipping path is preserved.

If someone would give me some advice on how to proceed it would be great.

**Comment by fmw42** [comment]
This should do what you want and antialias as well. Unix syntax

```
magick identify -quiet -format "%[8BIM:1999,2998:#1]" original.tif[0] |\
magick -quiet original.tif[0] +repage \
\( -density 96 - -channel rgb -negate -blur 0x1 -level 50x100% +channel \) \
-alpha off -compose copy_opacity -composite \
-define tiff:alpha=associated \
original_clip.tif
```

**Reply by stigsnor** [reply]
Thank you for taking the time to help me out.
Now it's time for me to dive into the documentation and understand what each part is actually doing.

You are awesome.

**Reply by stigsnor** [reply]
The code provided did solve my problem and I marked it as the answer.
But it turns out that the original command had a bug that will be fixed.
I'm removing your solution as the answer and setting it on the patch comment as it should have been working "out of the box".
I hope you don't find that to harsh as you are still awesome.

**Comment by snibgo** [comment]
The command and file given in the OP work fine for me, with v7.0.8-64 and Inkscape 1.0.

However, I suggest adding "+clip" before the write. It makes no difference in this case, but is good practice to "turn off" the clipping.

**Reply by stigsnor** [reply]
Thanks for looking into this

**Comment by fmw42** [comment]
First line gets the 8bim profile containing the clip path.
Second line reads the tif
Third line gets the clip path via pipe, makes an image, inverts it antialias its boundary
Fourth line puts it into the alpha channel
Fifth line sets the TIF alpha format
Sixth line saves the result

**Comment by snibgo** [comment]
Fred's command is good.

But the OP command should also be good. It works in v7.0.8-64, but apparently not in 7.0.10-13. This suggests a recent bug.

A developer may comment.

**Reply by fmw42** [reply]
Yes, the original command from Anthony fails for me also in IM 7.0.10-14 and has not worked for quite a few releases. Not sure when it stopped. For me it makes a totally transparent image.

**Reply by stigsnor** [reply]
I'm actually trying to use the wrapper Magick.NET. But as the conversion failed I wanted to see the commands working without the wrapper.

After testing versions it is working when Magick.NET is using ImageMagick 7.0.9-19, but it does not work when the wrapper is using ImageMagick 7.0.9-26 beta.

7.0.9-19 = works as expected
7.0.9-26 beta = fails

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by wymmeroo** [comment]
we have the same issue as @stigsnor has posted. 

we tried his image and command "**magick original.tif -alpha transparent -clip -alpha opaque output.png**" on a Windows 10 machine with version: ImageMagick 7.0.10-16 Q16 x64 

in version ImageMagick 7.0.10-15, these issue should be fixed, but we still face the same issue.
![image](https://user-images.githubusercontent.com/13727378/83721140-c31d1b00-a63a-11ea-874f-650d8a359a46.png)


can a developer comment on this?

@urban-warrior any ideas?

**Comment by mruf** [comment]
I can confirm that using the command and image @stigsnor posted still does not produce the expected image when using a self-built version of ImageMagick 7.0.10-16 on Ubuntu.
The box in the resulting image is transparent.
